### PR TITLE
[TxPool] Prune accounts with nonce holes during high memory usage

### PR DESCRIPTION
**Pull request by dbrajovic**
# Description

This PR introduces an asynchronous handler that is invoked at high gauge levels. Due to a possibility of some accounts having nonce holes ("stuck" enqueued transactions) taking up slots, the handler prunes each account satisfying this criterium. After pruning the handler goes into a short cooldown to avoid successive calls. 

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [ ] I have tested this code manually

# Additional comments

Fixes EDGE-732

### CODE REVIEW

**Review by zivkovicmilos — APPROVED** [review]
Looks great 💯

File: txpool/txpool_test.go [hunk 1]
```diff
@@ -2016,6 +2140,7 @@ func TestRecovery(t *testing.T) {
 }
 
 func TestGetTxs(t *testing.T) {
+	t.SkipNow()
```
Comment (0xAleksaOpacic) [inline txpool/txpool_test.go]: why?

**Review by 0xAleksaOpacic — APPROVED** [review]
LGTM

File: txpool/slot_gauge.go [hunk 2]
```diff
@@ -27,6 +31,12 @@ func (g *slotGauge) decrease(slots uint64) {
 	atomic.AddUint64(&g.height, ^(slots - 1))
 }
 
+// highPressure checks if the gauge level
+// is higher than the highPressureMark (0.8 * max)
+func (g *slotGauge) highPressure() bool {
+	return g.read() > uint64(highPressureMark*float64(g.max))
```
Comment (Kourin1996) [inline txpool/slot_gauge.go]: There may be an error depends on `highPressureMark` and `g.max`. For example, the error of right term is 1 where highPressureMark=0.29 and max=100. I think it's ok as far as highPressureMark is 0.8. I recommend you to calculate by integer division as long as highPressureMark can be represented in percent without the value after decimal point.

File: txpool/slot_gauge.go [hunk 3]
```diff
@@ -27,6 +31,12 @@ func (g *slotGauge) decrease(slots uint64) {
 	atomic.AddUint64(&g.height, ^(slots - 1))
 }
 
+// highPressure checks if the gauge level
+// is higher than the highPressureMark (0.8 * max)
```
Comment (Kourin1996) [inline txpool/slot_gauge.go]: Why do you write actual value (0.8) instead of highPressureMark?

**Review by Kourin1996 — COMMENTED** [review]
Just small comments

File: txpool/slot_gauge.go (hunk 2 above)
Comment (dbrajovic) [reply txpool/slot_gauge.go]: Can you write the suggestion as a code snippet ?

File: txpool/slot_gauge.go (hunk 3 above)
Comment (dbrajovic) [reply txpool/slot_gauge.go]: [Fixed](https://github.com/0xPolygon/polygon-edge/pull/689/commits/8529040d699d5df794073eb115f9f41d58514291)

File: txpool/txpool_test.go (hunk 1 above)
Comment (dbrajovic) [reply txpool/txpool_test.go]: [Reversed](https://github.com/0xPolygon/polygon-edge/pull/689/commits/6538db785bd659d094b25f65d6bbcfe4adf07aa9)

File: txpool/slot_gauge.go (hunk 2 above)
Comment (Kourin1996) [reply txpool/slot_gauge.go]:
```go
const (
     highPressureMark = 80 // 80%
)

func (g *slotGauge) highPressure() bool {
        return g.read() > (highPressureMark*g.max) / 100
}
```

File: txpool/txpool_test.go (hunk 1 above)
Comment (zivkovicmilos) [reply txpool/txpool_test.go]: @Aleksao998 ping

File: txpool/slot_gauge.go (hunk 2 above)
Comment (dbrajovic) [reply txpool/slot_gauge.go]: Great catch, I was not aware of [this behaviour](https://goplay.tools/snippet/RP60no2ZcyE) in Go.

(the `highPressureMark` is `const` anyhow, but maybe we'll make it configurable in the future)

[Refactored](https://github.com/0xPolygon/polygon-edge/pull/689/commits/e3eb589d38b2d4ff417f0a18db6cc50ff1e0bf48)

**Review by Kourin1996 — APPROVED** [review]

**Pull request merged**